compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/bir/codegen/JvmValueGen.java

```diff
@@ -680,8 +688,12 @@ private void createRecordInitWrapper(ClassWriter cw, String className, BIRNode.B
                 String refTypeClassName = getTypeValueClassName(typeRef.tsymbol.pkgID,
                         toNameString(typeRef));
                 mv.visitInsn(DUP2);
+
+                mv.visitInsn(ICONST_0);
+                mv.visitTypeInsn(ANEWARRAY, MAPPING_INITIAL_VALUE_ENTRY);
+
                 mv.visitMethodInsn(INVOKESTATIC, refTypeClassName, "$init",
-                        String.format("(L%s;L%s;)V", STRAND, MAP_VALUE), false);
+                        String.format("(L%s;L%s;[L%s;)V", STRAND, MAP_VALUE, BINITIAL_VALUE_ENTRY), false);
```

Comment:
I am in the process of removing `createRecordInitWrapper` method, can we please move (or duplicate) this in `createInstantiateMethod`

Reply:
We don't need initial values for referenced record types, so having the initial value population logic in "instantiate" should be enough. Moved to `createInstantiateMethod` in 6f89a47.